packages/server/test/hooks.spec.ts

```diff
@@ -311,6 +311,45 @@ describe('Hooks', () => {
         expect.anything(),
       );
     });
+    it('Should share the same context object reference across all hooks', (done) => {
+      let hook1Context: EvaluationContext;
+      let hook2Context: EvaluationContext;
+      let hook3Context: EvaluationContext;
+
+      client.getBooleanValue(FLAG_KEY, false, undefined, {
```

Comment:
Maybe we could also add some API/client level hooks and make sure they also use the same.